File: main_v1.0.0/BackEnd_v1.0.0/medknow-source-20260913/medknow/backend/src/main/java/com/yaozhidao/entity/Article.java
package com.yaozhidao.entity;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.LocalDateTime;

/** 科普文章 */
public class Article {

    private Long id;
    private String title;
    private String summary;
    private String content;
    private String coverImage;
    private String category;    // 中药/西药/养生/疾病/饮食
    private String author;
    private String authorTitle;
    private String tags;        // 逗号分隔
    private Integer readCount;
    private Integer likeCount;
    private Integer collectCount;
    private Integer shareCount;
    private Boolean status;     // 1=已发布
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime publishTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getCoverImage() { return coverImage; }
    public void setCoverImage(String coverImage) { this.coverImage = coverImage; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public String getAuthor() { return author; }
    public void setAuthor(String author) { this.author = author; }
    public String getAuthorTitle() { return authorTitle; }
    public void setAuthorTitle(String authorTitle) { this.authorTitle = authorTitle; }
    public String getTags() { return tags; }
    public void setTags(String tags) { this.tags = tags; }
    public Integer getReadCount() { return readCount; }
    public void setReadCount(Integer readCount) { this.readCount = readCount; }
    public Integer getLikeCount() { return likeCount; }
    public void setLikeCount(Integer likeCount) { this.likeCount = likeCount; }
    public Integer getCollectCount() { return collectCount; }
    public void setCollectCount(Integer collectCount) { this.collectCount = collectCount; }
    public Integer getShareCount() { return shareCount; }
    public void setShareCount(Integer shareCount) { this.shareCount = shareCount; }
    public Boolean getStatus() { return status; }
    public void setStatus(Boolean status) { this.status = status; }
    public LocalDateTime getPublishTime() { return publishTime; }
    public void setPublishTime(LocalDateTime publishTime) { this.publishTime = publishTime; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}

File: main_v1.0.0/FrontEnd_v1.0.0/MedKnow/app/src/main/java/com.medKnow.medknow/model/healthArticle/recommendedArticleList/Article.kt
package com.medKnow.medknow.model.healthArticle.recommendedArticleList

// 文章
data class Article (

    // 文章ID
    val articleId: Int,

    // 文章标题
    val title: String,

    // 文章摘要。前60个字符
    val summary: String,

    // 封面图片URL
    val coverImage: String,

    // 分类。中药 / 西药 / 养生 / 疾病 / 饮食
    val category: String,

    // 作者或来源
    val author: String,

    // 发布时间。格式yyyy-MMMM-dd HH:mm:ss
    val publishTime: String,

    // 阅读量
    val readCount: Int,

    // 点赞数
    val likeCount: Int,

    // 当前用户是否已点赞（未登录用户固定为 false）
    val isLiked: Boolean,

    // 当前用户是否已收藏（未登录用户固定为 false）
    val isCollected: Boolean

)